core/domain/question_services.py

```
@@ -464,6 +475,59 @@ def get_question_by_id(
+        question_id, question_version
+    )
+    try:
+        question_snapshot_model = (
+            question_models.QuestionSnapshotContentModel.get(
+                question_snapshot_id
+            )
+        )
+    except base_models.BaseModel.EntityNotFoundError as e:
+        raise QuestionSnapshotNotFoundError(
+            'Question snapshot for question %s version %s was not found.'
+            % (question_id, question_version)
```

Ruff (0.16.1)

[warning] Use format specifiers instead of percent format

Replace with format specifiers

(UP031)